src/Microsoft.Diagnostics.Runtime/src/DataTargets/ClrInfo.cs

```diff
@@ -97,8 +97,12 @@ internal ClrInfo(DataTarget dt, ClrFlavor flavor, ModuleInfo module, ulong runti
                     }
                 }
 
-                // We don't actually know the version of single file CLR since the version of the module would be the user's app version
                 Version = new Version();
+                if (info.IsClrRuntimeInfo2)
```

Comment:
Instead of reading twice, how about adding something like this to `ClrRuntimeInfo`?  (I did not test this code, please run it through your change...especially making sure the alignment is right.)

```C#

        public static unsafe bool TryReadClrRuntimeInfo(IDataReader reader, ulong address, out ClrRuntimeInfo header, out Version? version)
        {
            header = default;
            version = null;

            int size = sizeof(ClrRuntimeInfo);
            fixed (ClrRuntimeInfo* pHeader = &header)
                if (reader.Read(address, new Span<byte>(pHeader, size)) != size || !header.IsValid)
                    return false;

            if (header._version >= 2)
            {
                Span<int> intVersion = stackalloc int[4];
                Span<byte> buffer = MemoryMarshal.Cast<int, byte>(intVersion);

                if (reader.Read(address + (uint)size, buffer) == buffer.Length)
                    version = new Version(intVersion[0], intVersion[1], intVersion[2], intVersion[3]);
            }

            return true;
        }
```